### DISCUSSION TITLE
I got the esp32 s3 w display, Im stuck

### DISCUSSION THREAD

**Discussion by waltiti17** [Builty/TonexOneController#16] [General]
Hello, 

on waveshare I Press Boot, and plug the USB to PC, esp32 s3 is connected to com/port6

On the file I donwload = TonexOneController_V1.0.2.2_Waveshare_4_3_Display_BT_client  >  flash_download_tool_3.9.5.exe ,  I setup the little windows with esp32 s3 with  Factory / USB and  I press start, start to flash until finish.

I disconect the cable usb to pc, and reconect again and is a black screen on the waveshare.

Is flashed ok, or i wrong? 

Waveshare is only connected to PC with usb ,  May I need to connect to Tonex to boot Up ? 

I waiting for a power supply to can do  power supply >  Waveshare> Tonex

Please advice

Thanks 

![Screenshot 2024-11-19 121345](https://github.com/user-attachments/assets/7f99205b-a0bc-4494-8d65-0d59830c4daf)

**Comment by Builty** [comment]
It sounds like you did everything correct. When did you download the zip file from Github? I did have a bad version up there that had the wrong config and crashed on boot. Can you download the zip file again from the main branch and retry please.

The display does not require the Tonex for it to come on.
Is the hub powered? it might not have have enough current to run the board.

If none of that helps, please confirm exactly which Waveshare board you purchased.

**Comment by waltiti17** [comment]
Hello Builty

I open device manager, press/hold boot on wavshare, connect usb , on device manager apears com6
I download the file again TonexOneController-1.0.2.2.zip extract
go to releases v1.0.2.2 > 
TonexOneController_V1.0.2.2_Waveshare_4_3_Display_BT_client > extract
run flash_download_tool_3.9.5.exe
on the download tool windows I select ESP32 S3 / Factory/ USB 
On download panel I select com6
and press start, progress bar go to 100%, after some seconds show finish, close app, desconect USB, reconect and...

SCreen on, but black screen.

When I plug the unit first time to usb its boot up with software that come with the waveshare, looks like windows, more or less.

I tryu connecting with a 9v power and is the same screen turns on but is black screen, same as if I boot with USB powered from PC.

I got the waveshare from aliexpress may be thats why???? its come on a white a light blue box that say waveshare , looks the same.

I think I doing something wrong, not sure what. 

Any idea? 

I can return the unit, no big deal, but I was wishing this works

thanks

**Reply by Builty** [reply]
Everything you describe is exactly correct (apart from the black screen of course.)
I wonder if this board has less flash memory or ram or something else unusual.

I have created a special test build, in Releases/Special.
This one does not attempt to initialse the USB host, so it won't control the Tonex, however that means that we can now get access to the debug logs via this USB port.

Please try to flash this special release. After flashing, unplug the USB cable and then plug back in again. 
Then on your PC, run a program that can view a serial terminal. For example, the free program "Teraterm".
Run this program, and set it to the same serial port you found before. Set the baud rate to 115200.

My guess is that you will see crash logs appearing. Post them here and hopefully we can figure this out.
If you don't see anything, try pressing the touch screen and see if logs appear then.

It might also be worth opening the case and checking that all connectors are secure. It doesn't sound like its this though, as it worked when you got it.

A few others have successfully got this going on the Waveshare board, so its really strange that yours it not working.

**Comment by waltiti17** [comment]
Hello, Im sorry I didn't find the special release, Also i checked the zip TonexOneController-1.0.2.2

So I tried again here is the log my be you see something wrong? 

![Screenshot 2024-11-25 114219](https://github.com/user-attachments/assets/d2206995-4dcd-45cc-8055-25832f871954)

I mark with red some parts, I try the Start button, Start, and Start All. 

I flashed the unit 20 + times, new cables changed the USB port, screen turned on but is black, with no USB cable or power adapter. 

Something I doing wrong and I dont know what it is, I flashed lots of things, phones xbox, PS, name it, it's crazy

**Comment by Builty** [comment]
Do  you have the Github view set to the Main branch?

![image](https://github.com/user-attachments/assets/75d395c9-189c-4064-9ee7-ef1ccd654e37)

The fixed release 1.0.2.2 is committed to the main branch. The one on the release tag is still the bad one. 

The special release is also committed to this main branch.
![image](https://github.com/user-attachments/assets/3878e836-53e9-4bb5-9a7f-66363ecc8311)

If you can't see this special release, then there is something wrong with the Github view filter thing, which might explain why you are still seeing the bad release firmware.

**Comment by waltiti17** [comment]
Hello Builty/Greg,

I find the special release. Thanks a lot, 
I have Windows 11

For the file NoUSB_Debug_logger.zip, its BOOT UP the screen, GREAT, the touch screen doesn't work. Here is the Teraterm COM log 

![LOG NoUSB_Debug_logger](https://github.com/user-attachments/assets/a474fc35-cb2b-46e3-a81c-9ec94c2b2e4e)


For the file  TonexOneController_V1.0.3.1_Waveshare_Zero_BT_client.zip, NO BOOT, here is the Teraterm COM log 

![LOG TonexOneController_V1 0 3 1_Waveshare_Zero_BT_client](https://github.com/user-attachments/assets/6f130f03-2159-4b29-b170-92b6f7f5d59f)


For the file  TonexOneController_V1.0.2.2_Waveshare_4_3_Display_BT_client.zip , NO BOOT, like always, here is the Teraterm COM log

![LOG TonexOneController_V1 0 2 2_Waveshare_4_3_Display_BT_client](https://github.com/user-attachments/assets/bc9ce249-7e48-4782-b599-e03cee0f5743)


Hope this helps

Thanks a lot

**Comment by Builty** [comment]
This does help. So the processor is crashing on boot.
Can you try this please. In the flash download tool, press the Erase button.
And then write the flash again as normal, with the TonexOneController_V1.0.2.2_Waveshare_4_3_Display_BT_client 

I had some reports of boot crashing on the Zero. It seems the developmenrt environment I use erases the flash, but the flash programmer doesn't do that unless you ask. So it had some left over old memory set. I think it might be what is happening here.

Try the flash erase. I will also be releasing a new version soon that has changed the partition table to be closer to the normal setting for this board, which might also help.

**Comment by waltiti17** [comment]
Hello again,

Erase all, I did it before, and I do it again now, and NO, no boot-up, I will wait for the new release, sorry man, thanks a lot for the help

I don't understand why happened to me and no other users. Windows version, Waveshare version, flash tool version..... why meeeee ???? lol

Thanks again

**Comment by Builty** [comment]
Yeah things have been more trouble than I hoped it would be.
Seems part of the reason is that programming via the development IDE (Visual Studio Code) is not quite the same as doing it via the programming tool. A number of early users have been building it themselves and using the IDE. Or some using Mac and having to use command line, which is closer to the IDE.

I think we are almost there anyway. V1.0.3.1 beta has fixed boot issues for multiple people on the Zero.
I have created V1.0.3.2, just need to test all 5 releases fully to make sure I don't miss anything this time. 
Sorry to you for the troubles.

**Comment by waltiti17** [comment]
noo, thanks you for this, is awesome, hope I can make this works,

the only version that boots for me is  NoUSB_Debug_logger.zip

**Reply by Builty** [reply]
V1.0.3.2 is done, please give that a try. 
I did a lot more testing. Did things like erase the whole flash, and then programmed. In one case I had to program it twice, but seems like its all good now

**Comment by waltiti17** [comment]
YES YES YES YES YES !!!! works !!!!!! thanks a lot, I install the pedal version awesome. !!!!! works on the first flash try !!!!!

Also, Chocolate Work great, perfect for Stomp Mode on Tonex One, so you can turn on off with the Tonex One and dive all the 20 presets with the chocolate, great !!!

A question, there is a way that when you select a preset with the chocolate , 1 C for example, if you press again C on chocolate , its turn on , if press again turn it off, So i can " hide  " the Tonex One under the pedalboard, with the waveshare secreen and the chocolate will be enough.?

Also, is possible that on chocolate when its get to bank 5, that is the last bank, when I press C and B, go to bank 1, and vice versa if I in bank 1.  Unless IK adds / allows more than 20 presets in the future.

ty

**Reply by Builty** [reply]
Awesome!

With the first question, do you mean, if the same preset is activated a second time via the chocolate, that it then sets the Tonex to disabled/bypass?

For the second part, that would need something changed in the Chocolate pedal itself. It does have some configuration but I've never tried it.

**Comment by waltiti17** [comment]
Hi BuiltyGreg

Yes, I means, if on chocolate I select a preset and if I press again the same preset in chocolate  it can bypass the present and if I press again engage again, right now I doing that with Tonex one in stomp box, but will be very handy, instead of jump to tonex pedal for that, do it all with the chocolate bar. Will make the tonex one dont be needed on top of the pedalboard, could be hide under the pedalboard for example. 

Thanks

What you do is awesome, these are only small suggestions.

**Reply by Builty** [reply]
OK, I understand what you mean. It would be pretty easy to code that.
I'm not sure that it would suit everyone though. For example, if you had something like a Line-6 for effects that also sends a Midi command to set the preset, then it 2 songs in a row had the same preset, then the second song would end up in bypass mode.

I am thinking about how to handle these little special cases. Its too much maintenance to create different builds for all possible options.
I am thinking about using WiFi. Put the ESP32 in to access point mode for some time period after boot. Then there could be a web page where you select which options you want, and then it gets saved.
That could also simplify bluetooth server/client modes, where it could become 1 build with both and web page select for which is used.

**Comment by waltiti17** [comment]
Greg, if you think it is only worth it for me forget it !!!! or maybe press B and C to bypass ? 

I have an HX Effects and a TONEX Pedal, but I dont use MIDI so far, and I have a Tonex one for boost/distortion/gain. HX effects have really good boost/distortion/gain pedals, but im a double rabbit hole with the Tonex Pedal and Tonex one with so many amps, cabs, and pedals. 

I always play low budget amps and cabs ( like solid state Marshall 100 ), so never had access to good amps, cabs, etc, Now that I play at home only, Im old, I can try and play everything Amps Cabs Pedal, and I can make it sound like any artist, it depends on how I wake up, LOL,  I play with one amp or another depending on the mood, crazy !!!! I can use an EVH el34, or EVH III, dual rectifier, Diezel, Marshall, and the cabs/DI that change a lot the sound of everything, I love Marshall Cabs with other brand heads. wow. 

Forget about that, it could be handy only for people who use the Tonex One as boost/distortion/gain only. 

Another question, there is a way to load skins from the TF card of the waveshare screen  version? maybe that can free up some memory, not sure. 

Thanks for all your efforts, let me know if you want I to post a video of your stuff running on my side anywhere. I see somebody do it, but want to ask you first. 

thanks

**Reply by Builty** [reply]
I'm also old and only play at home! Nothing wrong with that ;)

The web page config fixes a few other issues also, and makes it so I have to create less builds. I have it about 70% done already.

Re Skins on TF, I did actually try that. The isue is that Waveshare decided to put the TF card chip select pin onto a special chip that allows for more IO, using the I2C bus.  Thats not really bad, except that the IO expander chip they choose has a nasty limitation. You can set all pins to inputs, or all pins to outputs, but not any mixture. Plus, accessing the I2C bus has to be protected via a mutex, and the touch screen is on the same bus and running in a timer.
That's a long way of saying, it's very difficult to make it work.

I did try loading all images from TF to PSRAM at boot, before the other tasks need to access the I2C bus, but that consumes too much memory. The GUI designer only supports PNG, and not JPG which would be smaller. Maybe I can do something here to get the files to jpg and save some space. I'd love to have both amp and pedal skins in one build.

As far as videos, I am happy for you to do whatever you like. It's fine to post, but also fine if you don't want to.

**Comment by Builty** [comment]
Code to do the bypass toggle is done. As is the new WiFi/web config.
I'll release a beta later in the week.

**Reply by waltiti17** [reply]
wow, hands down brother !!!!

**Reply by Builty** [reply]
I've released a beta for the Waveshare Zero, that has the option (in the new web confguration) to enable this mode where second setting of save preset toggles bypass mode:
https://github.com/Builty/TonexOneController/tree/main/Releases/V1.0.4.1

Could you possibly give it a try and report how it goes? Thanks

**Reply by waltiti17** [reply]
> wow, hands down brother !!!!

Greg, I dont have the Zero, I have the waveshare Screen , will works anyway on screen version to test it?

**Reply by Builty** [reply]
Ah OK. No the Zero version won't boot on the 4.3B.
I'll compile a beta of the 4.3B soon.

**Reply by Builty** [reply]
Beta for 4.3B with pedal skins is in the releases/beta folder now.

**Reply by waltiti17** [reply]
> Beta for 4.3B with pedal skins is in the releases/beta folder now.

works awesome !!!!!!!! thanks so much.

Question, with the web tool maybe there is a way to load and remove png skins?  you know there is a lot lot lot of pedals 

thanks so much, Greg!!!!

**Reply by Builty** [reply]
Great!
With the skins, not sure there. I have a few ideas on how to improve that side. The pedal version does have a lot of room for more, if I've missed some common ones.

**Reply by waltiti17** [reply]
Hey greg, hope you are ok,

Asking if is possioble and not too much work, add an option to flip the screen ? in my pedalboard will be great if the cables ( usb ) are in the other side. Only if is not a big thing. Thanks

**Reply by Builty** [reply]
Hi. Would the rotation be 90 degrees, or 180 degrees? 
90 Degrees would change it to portrait mode and the UI would need to be redesigned to suit.
180 degrees is probably OK, I can look at it.

**Reply by waltiti17** [reply]
Yes 180, no portrait, just only for the cables, but only if that does not make another thing worse, like using more memory or any other resource, it is working super fine, I get the midichief because I see can turn on off the new Tonex FX, but it is not a stable app, the app keeps close randomly. Thanks Greg

**Reply by Builty** [reply]
OK, I will see what I can do.
Do you mean that a new version of Midi Chief is able to control the effects, like Chorus/Reverb etc?

By the way, in a few days I will be releasing a new version that has full graphical (on the 4.3B) and Midi control for all platforms, for every parameter/effect that it has. Reverb/Chorus/Modulation/gain/EQ etc.

**Reply by vurguuz** [reply]
@Builty did I read that right? You were able to make a live editor of the settings on the device ahead of IK?

**Reply by Builty** [reply]
Yes I did get the live editor going. I'm working on a new video at the moment to demonstrate. Planning to release beta versions in the next few days.
There are a few limitations due to the method I had to use (as IK doesn't have an easy way to do it.) 
Parameter changes take about 1 second to apply. And the audio on the pedal does a very brief drop out, maybe a quarter of a second.

Even with these limitations, I think its really good.

**Comment by waltiti17** [comment]
Wow, I just see the video and download the new beta version, I will test it on the weekend, but looks AWESOME !!!!!!!

thanks a lot !!!!

**Comment by waltiti17** [comment]
WOW WOW WOW, I see the new version 1.07.2 AWESOME, you make Tonex One more user-friendly than Tonex Pedal !!!!!

Thanks a lot for the flip/rotate option.

So, what next ?????  
maybe a web version adding an screen for amp/pedal skins, for the Waveshare Nano or esp devkit c

Thanks a lot !!!!!

**Reply by Builty** [reply]
Thanks.
The idea of enhancing the web interface was suggested yesterday. Yeah could do the amp skins.

As for what's next, I'm still getting requests for USB Midi, so I think I will work on updating the IDF library to a newer one to get USB hub support.
